crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -4,46 +4,54 @@
 //! we define the way to calculate the protocol fee based on the configuration
 //! parameters.
 
-use crate::{
-    boundary::{self},
-    domain,
+use {
+    crate::{
+        boundary::{self},
+        domain,
+    },
+    anyhow::anyhow,
+    primitive_types::U256,
 };
 
 /// Constructs fee policies based on the current configuration.
 #[derive(Debug)]
 pub struct ProtocolFee {
-    policy: Policy,
+    policy: PolicyRaw,
     fee_policy_skip_market_orders: bool,
 }
 
 impl ProtocolFee {
-    pub fn new(policy: Policy, fee_policy_skip_market_orders: bool) -> Self {
+    pub fn new(policy: PolicyRaw, fee_policy_skip_market_orders: bool) -> Self {
         Self {
             policy,
             fee_policy_skip_market_orders,
         }
     }
 
     /// Get policies for order.
-    pub fn get(&self, order: &boundary::Order, quote: Option<&domain::Quote>) -> Vec<Policy> {
+    pub fn get(
+        &self,
+        order: &boundary::Order,
+        quote: Option<&domain::Quote>,
+    ) -> anyhow::Result<Vec<Policy>> {
```

Comment:
This function became messy pretty quickly.
Not sure if others agree, but I would be willing, for simplicity, to make the `quote` mandatory here, and handle the missing quote at the callsite. 

In practice, we will require from all orders to have quotes anyway.